src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+internal sealed class AsyncToSyncMethodTransformer : SyntaxTransformer {
+	public AsyncToSyncMethodTransformer( SemanticModel model ) : base( model ) { }

```

Comment:
Right now this is based on `SemanticModel`. It uses that to do stuff like look up types to check if they are `Task`. There's only a bit of that in here right now though.

If this becomes a performance issue I think it will be possible to revisit it, but I'm interested in other people's feelings.

This will run once per method concurrently in the incremental generator pipeline and there will be caching inside VS, but of course not on command line builds.

Reply:
I would personally run with it as-is. It will be hard to actually profile until we have usage too.

Given we're planning to use this a lot and for a while, it will eventually be important to figure out the most performant but functional thing though.

Reply:
Yeah. The cost will be zero to start because this (and the thing that will call it) only runs if you have something to transform, so it's at least an incremental cost (vs. our analyzers where you often pay a lot up-front)